pkg/sql/logictest/testdata/logic_test/mixed_version_range_tombstones

```diff
@@ -0,0 +1,85 @@
+# LogicTest: cockroach-go-testserver-upgrade-to-master
```

Comment:
The behavior in this test will be non-deterministic because we don't control which node adopts the job that actually writes the MVCC range tombstones. I thought there was a knob to control job adoption, but can't immediately find it -- perhaps @ajwerner knows. If we have such a knob, we should do two tests: one for 22.2 adoption and one for 23.1 adoption.

We should also add a test case verifying that we write them unconditionally on finalization.